src/main/java/io/github/syst3ms/skriptparser/util/color/Color.java

```diff
@@ -0,0 +1,170 @@
+package io.github.syst3ms.skriptparser.util.color;
+
+import java.util.regex.Pattern;
+
+/**
+ * Represents a color, written like #xxxxxx or with an RGB format.
+ * @author Mwexim
+ */
+public class Color {
```

Comment:
Because you can indeed have a certain amount of enum values, but the Color class itself should still be initiatable. Enum classes are non-constructable and would therefore limit our options, unless you know a work-around.